### PR TITLE
[Azure][database_account] Add missing region dimension, remove ourdated azure metrics

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## What does this PR do?

- add `Region` dimension
- remove `"MongoRequestsCount", "MongoRequestsInsert", "MongoRequestsDelete", "MongoRequestsQuery", "MongoRequestsUpdate"` metrics - there are no such metrics according to documentation - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-documentdb-databaseaccounts-metrics

FYI: used databaseaccounts metrics contain much more dimensions than I've added in this PR - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-documentdb-databaseaccounts-metrics

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by zmoog** [comment]
> Remove "MongoRequestsCount", "MongoRequestsInsert", "MongoRequestsDelete", "MongoRequestsQuery", "MongoRequestsUpdate" metrics - there are no such metrics according to documentation - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-documentdb-databaseaccounts-metrics

Yeah, I also tried to search for these metrics in the Metrics section of a test MongoDB instance, I could not find these metrics.

However, I have found something similar by enabling the "classic" metrics:

CleanShot 2023-08-23 at 17 38 29@2x

The metric name is not quite the same, and we have multiple metrics for each insert/delete/query/etc operation. These metrics use a different namespace, "Microsoft.DocumentDb/databaseAccounts/classic".

Since these metrics are not available in our target namespace, I believe it's okay to remove them from the configuration.

File: packages/azure_metrics/data_stream/database_account/agent/stream/stream.yml.hbs [hunk 1]
```diff
@@ -31,8 +31,27 @@ resources:
...
-          - name: ["AvailableStorage", "DataUsage","DocumentCount", "DocumentQuota", "IndexUsage", "MetadataRequests", "MongoRequestCharge", "MongoRequests", "MongoRequestsCount",
-                   "MongoRequestsInsert", "MongoRequestsDelete", "MongoRequestsQuery", "MongoRequestsUpdate","ProvisionedThroughput", "NormalizedRUConsumption"]
+          - name: ["AvailableStorage", "DataUsage","DocumentCount", "DocumentQuota", "IndexUsage", "MetadataRequests", "MongoRequestCharge", "NormalizedRUConsumption"]
+            namespace: "Microsoft.DocumentDb/databaseAccounts"
+            ignore_unsupported: true
+            timegrain: "PT5M"
+            dimensions:
+            - name: "DatabaseName"
+              value: "*"
+            - name: "Region"
```
[the remark this answers is not in the document]
Comment (zmoog) [reply packages/azure_metrics/data_stream/database_account/agent/stream/stream.yml.hbs:41]: I can't afford looking at this right now, but I don't want to lose track of this potential enhancement.

I added the issue https://github.com/elastic/integrations/issues/7511 to track it.

Thank you for the heads up!

**Review by zmoog — APPROVED** [review]

**Comment by tetianakravchenko** [comment]
> The metric name is not quite the same, and we have multiple metrics for each insert/delete/query/etc operation. These metrics use a different namespace, "Microsoft.DocumentDb/databaseAccounts/classic".

@zmoog thank you for the close review! I really missed this part. strange that I can't find `Microsoft.DocumentDb/databaseAccounts/classic` on the https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-documentdb-databaseaccounts-metrics in the list of all metrics


note: that the possible valued for the command_name is clightly different from the `MongoRequestsCount`, `MongoRequestsInsert`, `MongoRequestsDelete`, `MongoRequestsQuery`, `MongoRequestsUpdate` - there are more options now:

Screenshot 2023-08-22 at 15 38 50

File: packages/azure_metrics/data_stream/database_account/agent/stream/stream.yml.hbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tetianakravchenko) [reply packages/azure_metrics/data_stream/database_account/agent/stream/stream.yml.hbs:41]: thank you for the follow up issue!

**Comment by zmoog** [comment]
> strange that I can't find Microsoft.DocumentDb/databaseAccounts/classic on the https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-documentdb-databaseaccounts-metrics in the list of all metrics

I am not surprised, sometimes the Azure Docs are hard to navigate. There's a ton of docs, but the noise / signal ration is not great.

> note: that the possible valued for the command_name is slightly different from the MongoRequestsCount, MongoRequestsInsert, MongoRequestsDelete, MongoRequestsQuery, MongoRequestsUpdate - there are more options now

This `command_name` dimension is a neat idea.

**Pull request merged**